### ISSUE TITLE
[feature request] Research Article – Contributor Edit access

### ISSUE THREAD

**Issue by thisislawatts** [ONEARMY/community-platform#2144]
**Is your feature request related to a problem? Please describe.**
A way for the research contributors to edit Research (so we can make it more collaborative)

Requirements: 
* [x] Contributors can edit a Research document they have been added to this means;
  * [x] Editing the Research Description
  * [x] Editing existing Research Updates
  * [x] Adding new Research Updates
* [x] If a Contributor adds a new Research update that content should appear as authored by them.
* [ ] It is not possible for the Research Update or Research Description to be updated by two people at once. #2217

**Comment by thisislawatts** [comment]
Added [🚧 Blocked], until the question here around type of collaborative editing required have been answered so that we better understand the ask.

**Comment by davehakkens** [comment]
Can we try and unblock this one @thisislawatts?

I think we won't have multiple people adding new posts to the research at once. 
I do think we could have multiple people at the same time editing a step (for instance a contributor makes an edit to step 1 and at the same time another contributor to step 5)

That said I don't think it would happen much in the beginning as we roll it out softley. So to keep it simple we could also opt for >  single user editing at a time?

**Comment by thisislawatts** [comment]
Thanks for clarifying @davehakkens, I think that is a reasonable requirement for the first version.

**Comment by thisislawatts** [comment]
@davehakkens could I get some guidance on how to style an error state, when someone attempts to edit a bit of content that someone else is actively working on. 

![Screenshot 2023-03-26 at 20 17 51](https://user-images.githubusercontent.com/472589/227795889-89316c86-4d9a-4761-8249-adcf572d14ea.jpg)

**Comment by davehakkens** [comment]
oh wauw nice feature! 
Looks perfect like this i'd say 👌

How does it work, when someone has the "edit" screen of the research or step open this message appears for other contributors?

**Comment by thisislawatts** [comment]
@davehakkens thats the idea. This hasn't been implemented yet in https://github.com/ONEARMY/community-platform/pull/2172, I will probably push the work up in a separate PR to make things easier to review.
